beacon/sync/src/test/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/CommonAncestorTest.java

```diff
@@ -89,27 +156,40 @@ void shouldSearchCommonAncestorWithSufficientRemoteData()
             dataStructureUtil.randomBytes32());
 
     when(recentChainData.getHeadSlot()).thenReturn(currentLocalHead);
-    when(recentChainData.containsBlock(any())).thenReturn(true);
+    when(recentChainData.containsBlock(any())).thenReturn(false);
 
-    SafeFuture<UInt64> futureSlot =
+    final SafeFuture<UInt64> futureSlot =
         commonAncestor.getCommonAncestor(peer, firstNonFinalSlot, status.getHeadSlot());
 
     assertThat(futureSlot.isDone()).isFalse();
 
     verify(peer)
         .requestBlocksByRange(
             eq(syncStartSlot),
-            eq(CommonAncestor.BLOCK_COUNT),
+            eq(BLOCK_COUNT_PER_ATTEMPT),
+            blockResponseListenerArgumentCaptor.capture());
+
+    requestFuture.complete(null);
+
+    assertThatSafeFuture(
+            commonAncestor.getCommonAncestor(peer, firstNonFinalSlot, status.getHeadSlot()))
+        .isCompletedWithValue(firstNonFinalSlot);
+  }
+
+  private void verifyAndRespond(
+      final UInt64 syncStartSlot, final SafeFuture<Void> requestFutureAttempt) {
+    verify(peer)
+        .requestBlocksByRange(
+            eq(syncStartSlot),
+            eq(BLOCK_COUNT_PER_ATTEMPT),
             blockResponseListenerArgumentCaptor.capture());
 
     respondWithBlocksAtSlots(
-        requestFuture,
+        requestFutureAttempt,
         blockResponseListenerArgumentCaptor.getValue(),
         syncStartSlot,
-        CommonAncestor.BLOCK_COUNT);
+        BLOCK_COUNT_PER_ATTEMPT);
 
-    requestFuture.complete(null);
-    // last received slot is the best slot
-    assertThat(futureSlot.get()).isEqualTo(syncStartSlot.plus(CommonAncestor.BLOCK_COUNT).minus(1));
```

Comment:
seems important to check where we're at?

Reply:
mmm. I haven't fund where you think I missed the check.. additional hints?

Reply:
i just mean we've removed this validation? im not sure why we would remove it...

Reply:
No it isn't removed. It's just the diff looking weird.